### PR TITLE
Describe ItemCollection and CollectionCollection

### PR DESCRIPTION
**Pull request by emmanuelmathot**
**Related Issue(s):** #1251

**Proposed Changes:**

1. ItemCollection
2. CollectionCollection

**PR Checklist:**

- [X] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [X] This PR has **no** breaking changes.
- [ ] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md)
      **or** a CHANGELOG entry is not required.
- [ ] This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec),
      and I have opened issue/PR #XXX to track the change.

### CODE REVIEW

**Comment by emmanuelmathot** [comment]
@m-mohr sections are moved to READMEs but I am not sure to understand what you meant by context. Please review

File: item-spec/README.md [hunk 1]
```diff
@@ -21,3 +21,14 @@ schemas validate additional fields defined in *[Common Metadata](common-metadata
...
+## Generic Collections of Items
+
+In some cases, a set of Items must be grouped together in a unique file such as in the response
+of a [collection search query](https://github.com/radiantearth/stac-api-spec/tree/release/v1.0.0/ogcapi-features#collection-items-collectionscollectioniditems).
+When a set of Items must be grouped together in unique file, the [ItemCollection](https://github.com/radiantearth/stac-api-spec/blob/main/fragments/itemcollection/README.md)
+object must be used.
+
+Similarly to the relationship between a GeoJSON Feature and a STAC Item,
+a STAC ItemCollection should be a valid GeoJSON [FeatureCollection](https://tools.ietf.org/html/rfc7946#section-3.3)
+to allow interoperability with existing tools that support GeoJSON.
```
Comment (m-mohr) [inline item-spec/README.md:34]:
```suggestion
> \[!NOTE]
> In cases where a set of Items is grouped together (e.g. for providing search results),
> the [ItemCollection](https://github.com/radiantearth/stac-api-spec/blob/main/fragments/itemcollection/README.md)
object has been defined in the STAC API specification.
> 
> Similarly to the relationship between a GeoJSON Feature and a STAC Item,
> a STAC ItemCollection should be a valid GeoJSON [FeatureCollection](https://tools.ietf.org/html/rfc7946#section-3.3)
> to allow interoperability with existing tools that support GeoJSON.
```

File: collection-spec/README.md [hunk 2]
```diff
@@ -44,3 +44,10 @@ a collection and be able to adapt it to most any data model.
 
 Implementors are encouraged to do what makes sense for them, and to check out the [examples](../examples/) and 
 [other implementations](https://stacindex.org/catalogs) for current best practices.
+
+## Generic Collection of Collections
+
+In some cases, a set of Collections must be grouped together in a unique file such as in the response
+of the [collections search in a catalog](https://github.com/radiantearth/stac-api-spec/tree/release/v1.0.0/ogcapi-features#collections-collections).
+When a set of Items must be grouped together in unique file, the [CollectionCollection](https://github.com/radiantearth/stac-api-spec/blob/604ade6158de15b8ab068320ca41e25e2bf0e116/ogcapi-features/openapi-collections.yaml#L121)
+structure as defined in STAC API must be used.
```
Comment (m-mohr) [inline collection-spec/README.md:53]: I think I'd tone this down a bit:

```suggestion
> \[!NOTE]
> In cases where a set of Collections is grouped together (e.g. for providing search results),
> the [CollectionCollection](https://github.com/radiantearth/stac-api-spec/blob/604ade6158de15b8ab068320ca41e25e2bf0e116/ogcapi-features/openapi-collections.yaml#L121)
object has been defined in the STAC API specification.
```

In was wondering whether we should create a fragment similar to the ItemCollection an link to that...

Comment (emmanuelmathot) [reply collection-spec/README.md:53]: in STAC API spec?

Comment (m-mohr) [reply collection-spec/README.md:53]: Yes. Alongside the ItemCollection

Comment (emmanuelmathot) [reply collection-spec/README.md:53]: Ok will do the pr in api

Comment (m-mohr) [reply collection-spec/README.md:53]: Had some spare time, here's the PR https://github.com/radiantearth/stac-api-spec/pull/448 @emmanuelmathot 
Once merged we can update/add links to this PR.

**Comment by emmanuelmathot** [comment]
CollectionCollection defined here: https://github.com/radiantearth/stac-api-spec/pull/448

**Review by m-mohr — APPROVED** [review]

**Review by cholmes — APPROVED** [review]

**Pull request merged**